### PR TITLE
Add unit test for `notice_origin`

### PR DESCRIPTION
**Pull request by ericcheatham**
This PR also fixes a few code issues in the NoticeOrigin class

This PR addresses #103

Signed-off-by: Eric Cheatham ericjcheatham@gmail.com

### CODE REVIEW

File: tests/test_class_notice_origin.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        self.notice_hint = Notice("hint", "hint")
+
+        self.notices = [
+            self.notice_info,
+            self.notice_warn,
+            self.notice_errr,
+            self.notice_hint
+        ]
+
+        self.notice_origin = NoticeOrigin(origin_str='origin_str')
```
Comment (nishakm) [inline tests/test_class_notice_origin.py]: I don't think you need to specify the key and value here. `NoticeOrigin('origin_str')` is enough. Only use it when there are preset arguments. Eg: `def options(required, optional=None)` so when using this without the extra argument you will do `options('required')` and with the extra argument`options('required', optional='ok')`

File: classes/notice_origin.py [hunk 2]
```diff
@@ -50,10 +50,10 @@ def print_notices(self):
                 warnings = warnings + notice.message
             if notice.level == 'error':
                 errors = errors + notice.message
-            if notice.level == 'hints':
+            if notice.level == 'hint':
```
Comment (nishakm) [inline classes/notice_origin.py]: Good catch!

File: classes/notice_origin.py [hunk 3]
```diff
@@ -50,10 +50,10 @@ def print_notices(self):
...
                 hints = hints + notice.message
-        notice_msg = formats.notice_format(
-            origin=self.origin, info=info, warnings=warnings, errors=errors,
+        notice_msg = formats.notice_format.format(
+            origin=self.__origin_str, info=info, warnings=warnings, errors=errors,
```
Comment (nishakm) [inline classes/notice_origin.py]: Could you change this to `self.origin_str`? There is a property getter defined for it. Thanks!

**Comment by nishakm** [comment]
@ericcheatham Thanks! This is a lot of work and I appreciate it! Could you elaborate on your work in your commit message?
```
This PR also fixes a few code issues in the NoticeOrigin class <-- could you elaborate on the code issues? Mismatch in key 'hint', using incorrect property 'origin' etc?

This PR addresses #103 <-- replace with 'Resolves #103' 

Signed-off-by: Eric Cheatham <ericjcheatham@gmail.com>
```
There are a couple more changes needed in the file but other than that, the PR looks good.

When you update, please amend this commit rather than adding a new commit. Thanks again for your work!

File: tests/test_class_notice_origin.py (hunk 1 above)
Comment (ericcheatham) [reply tests/test_class_notice_origin.py]: I can change that. It's a bit of a force of habit for me. 👍

**Comment by nishakm** [comment]
Looks good to me 👍 Thanks again!

**Pull request merged**